File: backend/app/schemas/agent_monitoring.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AgentTraceResponse(BaseModel):
    """单条 AgentTrace 响应。"""

    id: int
    trace_id: str
    conversation_id: int | None
    user_id: int | None
    graph_name: str
    status: str
    input_message: str | None
    output_message: str | None
    total_tokens: int | None
    prompt_tokens: int | None
    completion_tokens: int | None
    tool_calls_count: int
    node_steps: int
    latency_ms: int | None
    started_at: datetime
    ended_at: datetime | None
    error_type: str | None
    error_message: str | None
    is_flagged: bool
    meta: dict[str, Any] | None

    model_config = ConfigDict(from_attributes=True)


class AgentTraceListResponse(BaseModel):
    """AgentTrace 列表分页响应。"""

    items: list[AgentTraceResponse]
    total: int

    model_config = ConfigDict(from_attributes=True)


class AgentSpanResponse(BaseModel):
    """单条 AgentSpan 响应。"""

    id: int
    trace_id: str
    parent_span_id: int | None
    span_type: str
    span_name: str
    status: str
    started_at: datetime
    ended_at: datetime | None
    latency_ms: int | None
    input_data: dict[str, Any] | None
    output_data: dict[str, Any] | None
    error_info: dict[str, Any] | None
    token_usage: dict[str, Any] | None
    meta: dict[str, Any] | None

    model_config = ConfigDict(from_attributes=True)


class AgentSpanListResponse(BaseModel):
    """AgentSpan 列表响应。"""

    items: list[AgentSpanResponse]

    model_config = ConfigDict(from_attributes=True)


class AgentTraceStatsResponse(BaseModel):
    """AgentTrace 统计概览。"""

    total: int
    today_count: int
    failed_count: int
    avg_latency_ms: int

    model_config = ConfigDict(from_attributes=True)


class AgentEvaluationResponse(BaseModel):
    """单条 AgentEvaluation 响应。"""

    id: int
    trace_id: str
    conversation_id: int | None
    eval_type: str
    dimension: str
    score: float
    reason: str | None
    evaluated_at: datetime
    evaluator_model: str | None
    meta: dict[str, Any] | None

    model_config = ConfigDict(from_attributes=True)


class AgentEvaluationListResponse(BaseModel):
    """AgentEvaluation 列表分页响应。"""

    items: list[AgentEvaluationResponse]
    total: int

    model_config = ConfigDict(from_attributes=True)


class AgentEvaluationStatsResponse(BaseModel):
    """AgentEvaluation 统计概览。"""

    total_evaluations: int
    avg_score: float
    low_score_count: int
    dimension_avgs: list[dict[str, Any]]

    model_config = ConfigDict(from_attributes=True)


class AgentEvaluationTrendResponse(BaseModel):
    """AgentEvaluation 趋势响应。"""

    items: list[dict[str, Any]]

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/agentMonitoring.ts
export interface AgentTrace {
  id: number
  trace_id: string
  conversation_id: number | null
  user_id: number | null
  graph_name: string
  status: string
  input_message: string | null
  output_message: string | null
  total_tokens: number | null
  prompt_tokens: number | null
  completion_tokens: number | null
  tool_calls_count: number
  node_steps: number
  latency_ms: number | null
  started_at: string
  ended_at: string | null
  error_type: string | null
  error_message: string | null
  is_flagged: boolean
  meta: Record<string, unknown> | null
}

export interface AgentSpan {
  id: number
  trace_id: string
  parent_span_id: number | null
  span_type: string
  span_name: string
  status: string
  started_at: string
  ended_at: string | null
  latency_ms: number | null
  input_data: Record<string, unknown> | null
  output_data: Record<string, unknown> | null
  error_info: Record<string, unknown> | null
  token_usage: Record<string, unknown> | null
  meta: Record<string, unknown> | null
}

export interface AgentEvaluation {
  id: number
  trace_id: string
  conversation_id: number | null
  eval_type: string
  dimension: string
  score: number
  reason: string | null
  evaluated_at: string
  evaluator_model: string | null
  meta: Record<string, unknown> | null
}

export interface AgentTraceListResponse {
  items: AgentTrace[]
  total: number
}

export interface AgentSpanListResponse {
  items: AgentSpan[]
}

export interface AgentTraceStatsResponse {
  total: number
  today_count: number
  failed_count: number
  avg_latency_ms: number
}

export interface AgentTraceListParams {
  skip?: number
  limit?: number
  conversation_id?: number
  user_id?: number
  status?: string
  is_flagged?: boolean
}

export interface AgentEvaluationListResponse {
  items: AgentEvaluation[]
  total: number
}

export interface AgentEvaluationStatsResponse {
  total_evaluations: number
  avg_score: number
  low_score_count: number
  dimension_avgs: Array<{
    dimension: string
    avg_score: number
  }>
}

export interface AgentEvaluationTrendResponse {
  items: Array<{
    date: string
    count: number
    avg_score: number
  }>
}

export interface AgentEvaluationListParams {
  skip?: number
  limit?: number
  dimension?: string
  min_score?: number
  max_score?: number
}
